specification/webpubsub/resource-manager/Microsoft.SignalRService/stable/2021-10-01/examples/WebPubSubHubs_CreateOrUpdate.json

```diff
@@ -0,0 +1,93 @@
+{
+  "parameters": {
+    "parameters": {
+      "properties": {
+        "eventHandlers": [
+          {
+            "properties": {
+              "urlTemplate": "http://host.com",
+              "userEventPattern": "*",
+              "systemEvents": [
+                "connect",
+                "connected"
+              ],
+              "auth": {
+                "type": "ManagedIdentity",
+                "managedIdentity": {
+                  "resource": "abc"
+                }
+              }
+            }
+          }
+        ]
```

Comment:
Nested resource must be created by its own PUT.

Reply:
The `eventHandler` in a `WebPubSubHub` has order. Re-order or replace all the `eventhandlers` in the `webpubsubhub` will be easier in such design. I think it's not an implicitly creation. And we'd like to work as [keyvault and accesspolicy](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/keyvault.json#L1253). KeyVault also show the content of accesspolicy and can update accesspolicy in parent resource's operation. How can we also achieve that?

Reply:
Then you should model event handlers as a property of webpubsubhub only not as a child resource. You can't have both.

The keyvault case is okay since access policies are a property of keyvault. (its [PUT api ](https://github.com/Azure/azure-rest-api-specs/blob/ce90f9b45945c73b8f38649ee6ead390ff6efe7b/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/keyvault.json#L302)does not comply to the RPC though).

Reply:
So, if event handler is just a property, can we have an api to update one event handler's properties only. I mean does ARM have such an api guidance to achieve this?

Reply:
PATCH is for partial update and it can update one event handler only. 
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource
https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc009-use-put-for-update-of-entire-resource-ie-replace-use-patch-only-for-partial-update-of-resource

You can't model a property of the parent resource as a child resource. They have different RBAC and you need to update the parent if a child resources are updated /deleted or vice versa.

Reply:
However, do you have an example to PATCH a list. From `json-merge` spec, it says `Also, it is not possible to patch part of a target that is not an object, such as to replace just some of the values in an array.` Currently, the structure is like:
```
"hub1" : {
    "eventhandlers": [
        {"name": "eventhandler1", "property1": "value1"},
        {"name": "eventhandler2", "property1": "value1"},
    ]
}
```
What's the correct way to only patch `eventhandler1`?
If we add an api like `PATCH /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalRService/webPubSub/{resourceName}/hubs/{hub}/eventHandler` Does it follow the spec?